app/js/reducers/artifacts.js

```diff
@@ -23,8 +23,11 @@ const artifacts = (state = initialState, action) => {
         });
         return state.set(action.artifact.id, nextState);
     }
-    case 'DELETE_ARTIFACT':
-        return state.delete(action.id);
+    case 'DELETE_ARTIFACT': {
+        // TODO, make it actually do something
+        const newState = ImmutableMap(state);
+        return newState.delete(action.id).toJS();
```

Comment:
Not sure if Immutable is being used properly. I'll have to do some digging, because this seems hacky (though working at the moment).